dalle2_pytorch/dalle2_pytorch.py

```diff
@@ -2057,7 +2058,7 @@ def sample(
 
         for unet_number, unet, vae, channel, image_size, predict_x_start, learned_variance in tqdm(zip(range(1, len(self.unets) + 1), self.unets, self.vaes, self.sample_channels, self.image_sizes, self.predict_x_start, self.learned_variance)):
 
-            context = self.one_unet_in_gpu(unet = unet) if is_cuda else null_context()
+            context = self.one_unet_in_gpu(unet = unet) if is_cuda and not distributed else null_context()
```

Comment:
One downside of using accelerate instead of handling ddp at a low level is that accelerate does not like when submodels are modified in any way. This becomes an issue in the sample method when a unet is moved to a device. In order to get around this, we create a parameter that prevents this move if we are doing distributed training.